### PR TITLE
Setting up the veto extension QS

### PR DESCRIPTION
**Pull request by LightGuard**
Creating the Veto example with tests

Basic tests in place to demonstrate how to use JPA entities, veto and
producers

### CODE REVIEW

File: cdi-veto/README.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+What is it?
+-----------
+
+[TODO: Finish / Update this]
+
+This project demonstrates a simple CDI Portable Extension and some of the SPI classes used
+to complete that task. This particular extension explores the ProcessInjectionTarget and
+InjectionTarget spi classes of CDI. To demonstrate a possible way to seed data into beans.
+
+A Portable Extension is essentially an extension to Java EE 6+ which is tailored to a specific
```
Comment (pmuir) [inline cdi-veto/README.md]: I think "Java EE 6" or above is clearer

File: cdi-veto/README.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+Target Product: EAP
+
+What is it?
+-----------
+
+[TODO: Finish / Update this]
+
+This project demonstrates a simple CDI Portable Extension and some of the SPI classes used
+to complete that task. This particular extension explores the ProcessInjectionTarget and
+InjectionTarget spi classes of CDI. To demonstrate a possible way to seed data into beans.
```
Comment (pmuir) [inline cdi-veto/README.md]: s/spi/SPI/

File: cdi-veto/README.md (hunk 1 above)
Comment (pmuir) [inline cdi-veto/README.md]: "is essentially" is redundent imo.

File: cdi-veto/README.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+A Portable Extension is essentially an extension to Java EE 6+ which is tailored to a specific
+use case which will run on any Java EE 6 or higher implementation. There may be something that the
```
Comment (pmuir) [inline cdi-veto/README.md]: s/There/This/ I think

File: cdi-veto/README.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+specifications don't support just yet, but could be implemented via a portable extension such as
+type safe messages or external configuration of beans.
+
+The project contains very simple domain model classes, an extension class, the service registration file
```
Comment (pmuir) [inline cdi-veto/README.md]: I'm not sure that service registration file is what the JAR specification calls service providers. Worth double checking.

File: cdi-veto/README.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+All you need to build this project is Java 6.0 (Java SDK 1.6) or better, Maven 3.0 or better.
+
+The application this project produces is designed to be run on JBoss Enterprise Application Platform 6 or JBoss AS 7.
+
+
+Configure Maven
+---------------
+
+Contributor: You can copy or link to the Maven configuration information in the README file in the root folder of the quickstarts. For example:
```
Comment (pmuir) [inline cdi-veto/README.md]: This is an instruction to you :-)

File: cdi-veto/src/main/java/org/jboss/as/quickstart/cdi/veto/CarManager.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+/**
+ * Producer for the {@link Car} entity.
+ */
+public class CarManager {
+    @Inject
+    private EntityManager em;
+
+    @Inject
+    UserTransaction utx;
```
Comment (pmuir) [inline cdi-veto/src/main/java/org/jboss/as/quickstart/cdi/veto/CarManager.java]: missing private keyword.

File: cdi-veto/src/main/java/org/jboss/as/quickstart/cdi/veto/CarManager.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+    private Logger log = Logger.getLogger(CarManager.class.getSimpleName());
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    /**
+     * Returns either a new instance or a managed instance of {@link Car}.
+     * Entities should be dependent scoped to avoid incompatible proxies between
```
Comment (pmuir) [inline cdi-veto/src/main/java/org/jboss/as/quickstart/cdi/veto/CarManager.java]: I think you need to say "The produced entity" or something to indicate you mean what comes out of this producer method.

File: cdi-veto/src/main/java/org/jboss/as/quickstart/cdi/veto/EntityManagerProducer.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+import javax.persistence.PersistenceContext;
+
+/**
+ * Simple resource producer for the {@link EntityManager}.
+ */
+public class EntityManagerProducer {
+    @Produces
+    @PersistenceContext
+    EntityManager em;
+}
```
Comment (pmuir) [inline cdi-veto/src/main/java/org/jboss/as/quickstart/cdi/veto/EntityManagerProducer.java]: missing private keyword.

**Comment by pmuir** [comment]
Great stuff, just the minor stuff I pointed out!

**Comment by sgilda** [comment]
Looks great! Pete, are you good with this?

**Comment by pmuir** [comment]
Yep.

**Pull request merged**